XenModel/XenAPI/VM.cs

```diff
@@ -3590,7 +3590,7 @@ public static void set_protection_policy(Session session, string _vm, string _va
         /// <param name="_value">The value</param>
         public static void set_snapshot_schedule(Session session, string _vm, string _value)
         {
-            session.proxy.vm_set_snapshot_schedule(session.uuid, (_vm != null) ? _vm : "", (_value != null) ? _value : "").parse();
+            session.proxy.vm_set_snapshot_schedule(session.uuid, (_vm != null) ? _vm : "", (_value != null) ? _value : Helper.NullOpaqueRef).parse();

```

Comment:
Sharad is working on this and has found the root cause and will be fixing it from XAPI, hence I have retained the original code here.